src/cycax/cycad/engines/cycax_part_freecad.py

```diff
@@ -224,43 +225,73 @@ def hole(
-        active_doc.activeView().viewAxometric()
-        active_doc.activeView().fitAll()
-        active_doc.activeView().saveImage(str(target_image_file), 2000, 1800, "White")
-        target_image_file = f"{target_path}-perspectiveTop.png"
+        view = view.upper()
+
+        if view is None:
+            view = "ALL"
+
+        self.change_view(active_doc=active_doc, side=view)
         FreeCADGui.SendMsgToActiveView("ViewFit")
-        active_doc.activeView().viewTop()
+
+        target_image_file = f"{self.filepath}-perspective{view}.png"
         active_doc.activeView().fitAll()
         active_doc.activeView().saveImage(str(target_image_file), 2000, 1800, "White")
 
-    def render_to_dxf(self, active_doc: App.Document, target_path: Path):
-        """This method will be used for creating a pdxf of the object currently in view.
+    def change_view(self, active_doc, side: str):
+        """This will change the gui view to show the specified side."""
+        
+        match side:
+            case "TOP":
+                active_doc.activeView().viewTop()
+            case "BACK":
+                active_doc.activeView().viewRear()
+            case "REAR":
+                active_doc.activeView().viewRear()
+            case "BOTTOM":
+                active_doc.activeView().viewBottom()
+            case "FRONT":
+                active_doc.activeView().viewFront()
+            case "LEFT":
+                active_doc.activeView().viewLeft()
+            case "RIGHT":
+                active_doc.activeView().viewRight()
+            case "ALL":
+                active_doc.activeView().viewAxometric()
+
+
+    def render_to_dxf(self, view: str|None = None):
+        """This method will be used for creating a dxf of the object currently in view.
         Args:
             active_doc: The active FreeCAD Gui
-            target_path: The Path the png needs to be saved under.
         """
+        view = view.upper()
+        active_doc = FreeCADGui.activeDocument()
+        if view is None:
+            view = "TOP"
+        self.change_view(active_doc=active_doc, side=view)
+        FreeCADGui.SendMsgToActiveView("ViewFit")
         __objs__ = []
         __objs__.append(active_doc.getObject("Shape"))
 
-        importDXF.export(__objs__, str(f"{target_path}-perspective.dxf"))
+        importDXF.export(__objs__, str(f"{self.filepath}-perspective{view}.dxf"))
 
-    def render_to_stl(self, active_doc: App.Document, target_path: Path):
+    def render_to_stl(self, active_doc: App.Document):
         """This method will be used for creating a STL of an object currently in view.
         Args:
             active_doc: The active FreeCAD Gui
-            target_path: The Path the png needs to be saved under.
         """
         for obj in active_doc.Objects:
             if obj.ViewObject.Visibility:
-                filename = f"{target_path}.stl"
+                filename = f"{self.filepath}.stl"
```

Comment:
We should add FreeCAD to the file name. Is that already part of self.filepath?

Reply:
If freecad it in the stl name then we will need to tell the assmebler that is it getting a freecad stl. I think this belongs in a different ticket.